python_modules/dagster/dagster/components/core/decl.py

```diff
@@ -82,7 +82,10 @@ class CompositePythonDecl(ComponentDecl[CompositeComponent]):
 
     def _load_component(self) -> "CompositeComponent":
         return CompositeComponent(
-            components={attr: decl._load_component() for attr, decl in self.decls.items()}  # noqa: SLF001
+            components={
+                attr: self.context.component_tree.load_component_at_path(decl.path)
```

Comment:
can maybe proxy through the context instead of having to go through tree prop in call sites 
ie `context.load_component_at_path`